server/src/modules/topics/topics.controller.ts

```diff
@@ -169,6 +174,16 @@ export class TopicsController {
         .json({ message: 'You do not have permission to delete this topic' })
     }
 
+    // Check if topic has posts
+    const posts = await this.postService.getPostsByTopic(topicId)
+    const hasPosts = posts.totalItems !== 0
+
+    if (hasPosts) {
+      return res
+        .status(StatusCodes.FORBIDDEN)
```

Comment:
```suggestion
        .status(StatusCodes.BAD_REQUEST)
```
403 (and the related 401) deals with permission-related issues, not so much the correctness of a request

Reply:
OK

Reply:
was debating which to use and tbh neither seemed like a perfect fit!
<img width="745" alt="image" src="https://user-images.githubusercontent.com/67887489/163766018-df5d186d-a75e-4999-82bf-b7424ac629b2.png">
<img width="493" alt="image" src="https://user-images.githubusercontent.com/67887489/163766044-b09414d8-3794-465f-8ecb-090b2c2f7f7e.png">

Reply:
done